src/main/java/UseCase/DatabaseManager.java

```diff
@@ -0,0 +1,184 @@
+        HashMap<String, UserAccount> d = this.database.get_data();
+        if (d.isEmpty()) {
+            d.put(username, account);
+            return true;
+        } else {
+            if (d.containsKey(username)) {
+                return false;
+            } else {
+                d.put(username, account);
+                return true;
+            }
+        }
+    }
+
+    public UserAccount retrieve_user_account(String username) {
+        return this.database.get_data().get(username);
+    }
+
+    public boolean is_pair_exist(UserDatabase database, String username, String password) {
+        if (database.get_data().containsKey(username)) {
+            if (database.get_data().get(username).equals(password)) {
+                return true;
+            } else {
+                return false;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    public boolean isUsernameExist(String username) {
+        if (this.database.get_data().containsKey(username)) {
+            return true;
+        }
+        return false;
+    }
+
+    public int number_user() {
+        return this.database.get_data().size();
+    }
+
+    public String[] list_username() {
+        return this.database.get_data().keySet().toArray(new String[0]);
+    }
+
+    public Object[][] writeDatabase() {
+        Object[][] userProfile = {{}};
+        String[] usernames = list_username();
+        for (int i = 0; i < number_user(); i++) {
+            UserAccount user = retrieve_user_account(usernames[i]);
+            Object[] userProfileEach = new Object[]{user.get_username(),
+                                                    user.get_full_name(),
+                                                    user.get_age(),
+                                                    user.get_pronouns(),
+                                                    user.get_location().get("Country"),
+                                                    user.get_location().get("Province"),
+                                                    user.get_location().get("City"),
+                                                    user.get_gender(),
+                                                    user.get_sexuality(),
+                                                    user.get_interests(),
+                                                    user.get_password()};
+            userProfile = Arrays.copyOf(userProfile, userProfile.length + 1);
+            userProfile[userProfile.length - 1] = userProfileEach;
+        }
+        return userProfile;
+    }
+}
+//
+//    private String[] dataFromCSV;
+//
+//    public DatabaseManager() {
+//        DataAccessInterface dataAccessInterface = new DataAccess();
+//        this.dataFromCSV = dataAccessInterface.read_csv();
+//    }
+//
+//    public UserDatabase createDatabase() {
+//        String splitBy = ",";
+//        UserDatabase database = new UserDatabase();
```

Comment:
Beware having huge comment blocks like this. It can be a bit confusing; make sure you delete these once you're sure that you don't have to reference them anymore.